### DISCUSSION TITLE
Кнопки в forn

### DISCUSSION THREAD

**Discussion by Mr-Coquin** [moonshine-software/moonshine#1052] [Q&A]
Тема заезженная с Button. Но не могу уже пару дней разобраться. Как в Moonshine на странице formPage можно работать с кнопками, в данном случаи интересуют такая возможность. 

**1 возможность** 
В низу формы есть дефолтная кнопка "Save" как  с права от ее добавить в таком же стиле скажем 3 свои кнопки 
![image](https://github.com/moonshine-software/moonshine/assets/123034940/9657ef26-98e0-4bb3-92ae-699a6496d846)

У меня получилось только через Layer но они тогда все находятся ниже данной кнопки.  Или же можно ее как то убрать вовсе и вынести эту кнопку "Save" в Layer под формой  через ActionButton ? Но тогда встает вопрос повесить событие отправки формы ?

**2 возможность**
Касательно самой кнопки "Save" как можно ее логику изменить:
1)Так что бы после ее нажатия форма сохранялась и перенаправляло автоматически на добавления нового элемента
2)При этом перенаправления передать сохранённые данные предыдущей формы ( в моем случаи это нужно для того, что бы пользователю часть информации подставлялась из прошлого сохранения для экономии времени при воде) 

Было бы классно получить пример как это сделать и можно ли вообще это реализовать максимально встроенными возможностями не прибегая к написанию своего компонента

**Comment by lee-to** [comment]
Вам нужно опубликовать страницы ресурса и в FormPage реализовать метод formComponent, он возвращает FormBuilder
необходимо добавить метод buttons и в него передать нужные дополнительные ActionButton

**Comment by lee-to** [comment]
Я кстати добавлю вам быстрый метод чтобы с ресурса туда кнопки отправить

**Reply by Mr-Coquin** [reply]
Это было бы круто

**Comment by Mr-Coquin** [comment]
А по 2-ой возможности можно как то сделать ? Пока только приходит в голову input hidden

**Reply by lee-to** [reply]
для это есть redirectAfter в доке об этом все есть, есть свойство и есть метод для более сложной кастомизации

**Reply by Mr-Coquin** [reply]
> для это есть redirectAfter в доке об этом все есть, есть свойство и есть метод для более сложной кастомизации

Ага, да вижу: https://moonshine-laravel.com/docs/resource/models-resources/resources-index#redirects
Вроде так верно ? 

    public function redirectAfterSave(): string
    {
        $item = $this->getItem();
       
        return $this->formPage()->url() . '/?' . http_build_query([
                'old_id' => $item->getKey()
            ]);
    }
    
  Или есь более изящный метод ?:)

**Reply by lee-to** [reply]
есть ощущение что то что вам нужно будет рабоать и просто через свойство protected ?PageType $redirectAfterSave = PageType::FORM;

**Reply by Mr-Coquin** [reply]
> есть ощущение что то что вам нужно будет рабоать и просто через свойство protected ?PageType $redirectAfterSave = PageType::FORM;

Нет, этот способ я знаю и я через его работаю обычно для перенаправления. Но тут мне нужно передавать около 4 параметров с предыдущего элемента для авто-выбору по default в новой форме, т.е. я скажем выбрал при создание нового элемента в ресурсе четыре параметра посредством select  из 10 (остальные 6 text-которые всегда пишутся руками при заполнение) , нажимаю "save" он сохраняет , перенаправляет меня на форму создания нового элемента и автоматически берет эти 4 значения из предыдущего, которое я до этого создал элемента. Что то такое получается тогда 

return $this->formPage()->url() . '/?' . http_build_query([
                 'old_id' => $item->getKey()
                'select1' => $item->select1,
 'select2' => $item->select2,
 'select3' => $item->select3,
 'select4' => $item->select4,
            ]);

**Reply by lee-to** [reply]
Ну тогда через метод, все верно
Кстати решение по кнопка уже готово https://github.com/moonshine-software/moonshine/pull/1055

**Reply by Mr-Coquin** [reply]
> Ну тогда через метод, все верно Кстати решение по кнопка уже готово #1055

Шикарно :) Спасибо
